kv.go

```diff
@@ -389,14 +405,25 @@ func (kv *kvs) get(key string) (KeyValueEntry, error) {
 	b.WriteString(kv.pre)
 	b.WriteString(key)
 
-	m, err := kv.js.GetLastMsg(kv.stream, b.String())
+	var m *RawStreamMsg
+	var err error
+	if revision == kvLatestRevision {
+		m, err = kv.js.GetLastMsg(kv.stream, b.String())
+	} else {
+		m, err = kv.js.GetMsg(kv.stream, revision)
+	}
+
 	if err != nil {
 		if err == ErrMsgNotFound {
 			err = ErrKeyNotFound
 		}
 		return nil, err
 	}
 
+	if m.Subject != b.String() {
```

Comment:
This check is only needed if you do line 413.

Reply:
Yeah... would the preference be to duplicate the error checking on 416 under the `if/else` and only check subject there? Also thinking would you want a different error for the case where someone provides a sequence that doesn't match subject?

Reply:
Comments addressed